### ISSUE TITLE
Decide and explicitly document if a featuregate can be "sunset" (removed/disabled).

### ISSUE THREAD

**Issue by bogdandrutu** [open-telemetry/opentelemetry-collector#7044]
Right now we have a debate if the featuregate should be removed, or we should keep them in a terminal state (Stable/Deprecated) forever (also allowing them to be changed forever since otherwise we still break functionality).

Bogdan's proposal:

Feature gates are to enable experimental features, and if users are using them they must not be consider from our "stable" interface. Hence we should allow moving a featuregate into a final state (Stable/Deprecated) and disallow setting it.

See some debate [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17981). This issue is to also decide if we are not ok with "Bogdan's proposal".

**Comment by codeboten** [comment]
I agree that feature gates should be removed eventually. It doesn't make sense to keep feature gates for things that don't do anything around.

**Comment by djaglowski** [comment]
> I agree that feature gates should be removed eventually. It doesn't make sense to keep feature gates for things that don't do anything around.

+1

**Comment by TylerHelmuth** [comment]
+1.  Once a featuregate is Stable/Deprecated for long enough the code should be allowed to be removed.  If we don't want that removal to be a jarring breaking change, maybe there is a step to error on startup when setting a Stable/Deprecated featuregate with a clear message?

**Comment by bogdandrutu** [comment]
@codeboten or @MovieStoreGuy one of you can open a PR and document this decision? Maybe add some requirements in terms of "must be at least foo versions in stable or deprecated state".

**Comment by MovieStoreGuy** [comment]
> @codeboten or @MovieStoreGuy one of you can open a PR and document this decision? Maybe add some requirements in terms of "must be at least foo versions in stable or deprecated state".

Makes sense to me

**Comment by codeboten** [comment]
Done here: https://github.com/open-telemetry/opentelemetry-collector/pull/7419

**Comment by MovieStoreGuy** [comment]
Thank you @codeboten !
